dspace-server-webapp/src/main/java/org/dspace/app/rest/ShibbolethRestController.java

```diff
@@ -53,8 +55,20 @@ public void shibboleth(HttpServletResponse response,
         if (redirectUrl == null) {
             redirectUrl = configurationService.getProperty("dspace.ui.url");
         }
-        log.info("Redirecting to " + redirectUrl);
-        response.sendRedirect(redirectUrl);
+
+        // Validate that the redirectURL matches either the server or UI hostname. It *cannot* be an arbitrary URL.
+        String redirectHostName = Utils.getHostName(redirectUrl);
+        String serverHostName = Utils.getHostName(configurationService.getProperty("dspace.server.url"));
+        String clientHostName = Utils.getHostName(configurationService.getProperty("dspace.ui.url"));
+        if (StringUtils.equalsAnyIgnoreCase(redirectHostName, serverHostName, clientHostName)) {
```

Comment:
@tdonohue since we are in shibboleth context, this is what I did in the current Shibboleth SLO PR:

I've added a specific method to that particular authentication method that retrieves a validated URL. To only allows redireccion for what is only configured.
```

    public static String getURLFromLogoutActionRequest(HttpServletRequest request) throws ShibAuthenticationException {
        String returnURL = null;
        // verify if we have shibboleth parameters (action and return)
        String action = request.getParameter("action");
        String shibLogoutURL = request.getParameter("return");

        // is shibboleth action for logout?
        if (ShibAuthentication.SHIBBOLETH_LOGOUT_ACTION.equals(action) && StringUtils.isNotBlank(shibLogoutURL)) {
            // for security issues we still need to validate return param
            ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
            String serverUrl = Utils.getBaseUrl(configurationService.getProperty("dspace.server.url"));
            serverUrl = getSecureURL(serverUrl, request);
            if (!shibLogoutURL.startsWith(serverUrl)) {
                throw new ShibAuthenticationException("Invalid 'return' param");
            }
            returnURL = shibLogoutURL;
        }
        return returnURL;
    }

```
Meanwhile, I did some refactoring and because of higher priority stuff and I bug I found with logout I didn't got the chance to rebase my work. 

This work it was based (I think) on @atarix83 work with Shibboleth authentication PR. He added a methods I think it was Utils.getBaseUrl to retrieve the correct and absolute URL. I think, since we are at Shibboleth context it makes sense to try to use those methods to validate the URL redirection.

Reply:
@paulo-graca : I'm not seeing much difference between your method and mine?  In mine, I'm doing a Hostname comparison to ensure the `redirectUrl` param matches the Hostname of either `dspace.server.url` or `dspace.ui.url`.  In your code pasted above, it looks like you are just making sure the `redirectUrl` *starts with* the same base URL as `dspace.server.url` (which is almost the same as a Hostname, without the HTTP/HTTPS).

So, I'm not sure how to respond here.  Did you want me to make changes to the code I've added to `ShibbolethRestController` to no longer validate against the Hostname, but check more of the URL?